Enhance budget and transaction management features

affan880 opened the pull request:
- Implement budget creation, listing, updating, and deletion with server integration.
- Refactor budget and transaction stores to improve state management and API interactions.
- Introduce loading states and error handling in various components for better user experience.
- Update UI components to reflect new data and improve overall layout.

affan880 committed:
Enhance budget and transaction management features

- Implement budget creation, listing, updating, and deletion with server integration.
- Refactor budget and transaction stores to improve state management and API interactions.
- Introduce loading states and error handling in various components for better user experience.
- Update UI components to reflect new data structures and improve overall layout.

affan880 committed:
fixed issues

affan880 committed:
Refactor CORS configuration and update cookie settings

kayoMichael commented on apps/server/src/controllers/auth.controller.ts:
```diff
@@ -162,7 +162,7 @@ export const logInUser = async (
       httpOnly: true,
       secure: env.NODE_ENV === 'production',
-      sameSite: env.NODE_ENV === 'production' ? 'strict' : 'lax',
+      sameSite: env.NODE_ENV === 'production' ? 'none' : 'lax',
```
we shouldn't do none here? We do have the same site domain in production.
fin-track.live
and api.fin-tracklive.

kayoMichael commented on apps/server/src/config/cors.ts:
```diff
@@ -1,14 +1,13 @@
+  origin: process.env.CLIENT_HOST === 'ALLOW_ALL' 
+    ? true 
+    : process.env.CLIENT_HOST 
+      ? [process.env.CLIENT_HOST] 
```
This is wrong logic for running prod environment locally since the frontend becomes localhost:80. Thus falling back to 5173 is going to fail since client_host is not included in that environment when running locally.

affan880 replied:
okay, I'll fix it

affan880 committed:
Fixed Endpoint Mismatch

affan880 committed:
Fixed

affan880 committed:
Update API endpoints to accept user ID for fetching categories and transactions

- Modified LIST_BY_USER endpoints for both categories and transactions to accept a user ID as a parameter.
- Updated the corresponding store files to utilize the new endpoint structure.

kayoMichael reviewed (approved):
LGTM. Good job on the budgets. Looks good
